tests/test_schema.py

```diff
@@ -17,22 +17,78 @@
 
 # This file contains tests for the schema for the yaml file
 
+
+import os
 from pathlib import Path
 
+import jsonschema
+
 import pytest
 
-from pytest_workflow.schema import validate_schema
+from pytest_workflow.schema import FileTest, WorkflowTest, validate_schema, \
+    workflow_tests_from_schema
 
 import yaml
 
-
-valid_yaml_dir = (Path(__file__).parent / Path("yamls") / Path("valid"))
-valid_yamls = [
-        (valid_yaml_dir / Path("valid_test.yml"))
-    ]
+valid_yaml_dir = Path(__file__).parent / Path("yamls") / Path("valid")
+valid_yamls = os.listdir(valid_yaml_dir.__str__())
 
 
 @pytest.mark.parametrize("yaml_path", valid_yamls)
 def test_validate_schema(yaml_path):
-    with yaml_path.open() as yaml_fh:
+    with Path(valid_yaml_dir / Path(yaml_path)).open() as yaml_fh:
         validate_schema(yaml.safe_load(yaml_fh))
+
+
+def test_WorkflowTest():
+    with (valid_yaml_dir / Path("dream_file.yaml")).open() as yaml_fh:
+        test_yaml = yaml.safe_load(yaml_fh)
+        tests = [WorkflowTest.from_schema(x) for x in test_yaml]
```

Comment:
You could parametrize this.

Reply:
But I am only testing one specific file... Will parametrize when I test more than one file.

Reply:
You have multiple test cases in the same file. With parametrization, it would be immediately obvious which one fails, if there is a failure.